components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -449,6 +478,42 @@ func generateVirtualService(instance *v1beta1.Notebook) (*unstructured.Unstructu
 			"timeout": "300s",
 		},
 	}
+
+	// Create a template map for the headers section
+	// of the Istio VirtualService
+	headers := map[string]interface{}{
+		"request": map[string]interface{}{
+			"set": map[string]interface{}{},
+		},
+	}
+
+	// whitelist for request headers that are not allowed to be set
+	whitelist := []string{"X-Forwarded-Proto", "X-RStudio-Request"}
```

Comment:
@DavidSpek as mentioned in https://github.com/kubeflow/kubeflow/pull/5660#discussion_r592294749 you don't need to worry about specifying specific headers.

When parsing the full list of provided headers from `notebooks.kubeflow.org/http-request-headers-set` just keep and set those that start with `X-`